src/main/scala/org/ergoplatform/explorer/db/models/composite/FlatBlock.scala

```diff
@@ -0,0 +1,118 @@
+package org.ergoplatform.explorer.db.models.composite
+
+import cats.MonadError
+import cats.instances.try_._
+import cats.syntax.functor._
+import org.ergoplatform.ErgoAddressEncoder
+import org.ergoplatform.explorer.Address
+import org.ergoplatform.explorer.db.models._
+import org.ergoplatform.explorer.protocol.models.{ApiBlockTransactions, ApiFullBlock}
+import org.ergoplatform.explorer.protocol.utils
+import org.ergoplatform.explorer.settings.ProtocolSettings
+
+import scala.util.Try
+
+/** Flattened representation of a full block from
+  * Ergo protocol enriched with statistics.
+  */
+final case class FlatBlock(
+  header: Header,
+  info: BlockInfo,
+  extension: BlockExtension,
+  adProofOpt: Option[AdProof],
+  txs: List[Transaction],
+  inputs: List[Input],
+  outputs: List[Output],
+  assets: List[Asset]
+)
+
+object FlatBlock {
+
+  def fromApi[F[_]: MonadError[*[_], Throwable]](
```

Comment:
Is it possible to avoid using effects in the model, so that model doesn't have dependency on cats?

Reply:
We can use concrete types here, `Try` for instance, but we're sticking to TF so there is no point in doing this. Moreover, it would make the code inconsistent and force us to always `.fold` on the result of this methods to use them with other code.